File: test/fixtures/sample.kt
package com.example.app

import com.example.models.User
import com.example.utils.Logger

data class Config(
    val host: String,
    val port: Int,
    val debug: Boolean = false
)

interface Repository<T> {
    fun findById(id: String): T?
    fun save(entity: T): T
    fun delete(id: String)
}

class UserRepository(private val config: Config) : Repository<User> {
    private val logger = Logger("UserRepository")

    override fun findById(id: String): User? {
        logger.info("Finding user: $id")
        return null
    }

    override fun save(entity: User): User {
        logger.info("Saving user: ${entity.name}")
        return entity
    }

    override fun delete(id: String) {
        logger.info("Deleting user: $id")
    }

    fun findByEmail(email: String): User? {
        return null
    }

    companion object {
        fun create(config: Config): UserRepository {
            return UserRepository(config)
        }

        fun default(): UserRepository {
            return UserRepository(Config("localhost", 5432))
        }
    }
}

object AppRegistry {
    private val services = mutableMapOf<String, Any>()

    fun register(name: String, service: Any) {
        services[name] = service
    }

    fun get(name: String): Any? {
        return services[name]
    }
}

fun initializeApp(config: Config): UserRepository {
    val repo = UserRepository.create(config)
    AppRegistry.register("userRepo", repo)
    return repo
}

val DEFAULT_CONFIG = Config("localhost", 8080)

File: test/fixtures/sample.scala
package com.example.app

import scala.collection.mutable
import scala.concurrent.Future

// Singleton object
object AppConfig {
  val defaultPort: Int = 8080
  var debug: Boolean = false

  def load(path: String): AppConfig = {
    new AppConfig(path)
  }
}

// Class with constructor
class AppConfig(val configPath: String) {
  val settings: mutable.Map[String, String] = mutable.Map.empty

  def get(key: String): Option[String] = settings.get(key)

  def set(key: String, value: String): Unit = {
    settings(key) = value
  }
}

// Trait definition
trait Repository[T] {
  def findById(id: Long): Option[T]
  def save(entity: T): Unit
  def delete(id: Long): Boolean
}

// Case class (value type)
case class User(id: Long, name: String, email: String)

// Class implementing trait
class UserRepository extends Repository[User] {
  private val store: mutable.Map[Long, User] = mutable.Map.empty

  override def findById(id: Long): Option[User] = store.get(id)

  override def save(entity: User): Unit = {
    store(entity.id) = entity
  }

  override def delete(id: Long): Boolean = {
    store.remove(id).isDefined
  }
}

// Type alias
type UserId = Long

// Val and var definitions
val MaxRetries: Int = 3
var currentUser: Option[User] = None

// Top-level function
def createRepository(): Repository[User] = {
  new UserRepository()
}
